### ISSUE TITLE
Cannot read all speech history in NVDA 2020.2 beta

### ISSUE THREAD

**Issue by brunoprietog** [jscholes/nvda-speech-history#13]
I'm using version 2020.2 of the add-on. It had been working well in NVDA 2020.1 until I tested the NVDA 2020.2 beta. Now as I scroll through the history, it repeatedly fails to read some items.

**Comment by jscholes** [comment]
Sorry to hear the add-on is giving you trouble, but there's very little I can do about this unless you tell me which items you're expecting to be read which are not, and the steps to reproduce the issue.

**Comment by brunoprietog** [comment]
Thanks for the answer. For example, in a UWP application, mail, if you tab through the items and then check through the history with Shift+F11, sometimes NVDA doesn't speak when you go through certain items that it should pronounce. Another case where it fails is browsing the desktop, if you browse through several items and then check them using Shift+F11 and Shift+F12, it only reads the last item when it reaches it with Shift+F12, the rest it doesn't read.

Here's the log for that specific part, the last test case I mentioned. Supposedly NVDA says so, but in reality it doesn't.

```
IO - inputCore.InputManager.executeGesture (18:10:41.439) - winInputHook (10632):
Input: kb(laptop):windows+d
IO - speech.speak (18:10:41.575) - MainThread (5080):
Speaking ['FolderView', 'lista', CancellableSpeech (still valid devInfo( isCanceledCache: False, isValidCallback: True, isValidCallbackDevInfo:  )]
IO - speech.speak (18:10:41.579) - MainThread (5080):
Speaking ['Microsoft Edge', '4 de 9', CancellableSpeech (still valid devInfo( isCanceledCache: False, isValidCallback: True, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:42.818) - winInputHook (10632):
Input: kb(laptop):downArrow
IO - speech.speak (18:10:42.853) - MainThread (5080):
Speaking ['NVDA', '5 de 9', CancellableSpeech (still valid devInfo( isCanceledCache: False, isValidCallback: True, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:42.994) - winInputHook (10632):
Input: kb(laptop):downArrow
IO - speech.speak (18:10:43.027) - MainThread (5080):
Speaking ['WinSCP', '6 de 9', CancellableSpeech (still valid devInfo( isCanceledCache: False, isValidCallback: True, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:43.140) - winInputHook (10632):
Input: kb(laptop):downArrow
IO - speech.speak (18:10:43.168) - MainThread (5080):
Speaking ['Microsoft Teams', '7 de 9', CancellableSpeech (still valid devInfo( isCanceledCache: False, isValidCallback: True, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:43.308) - winInputHook (10632):
Input: kb(laptop):downArrow
IO - speech.speak (18:10:43.339) - MainThread (5080):
Speaking ['Visual Studio Code', '8 de 9', CancellableSpeech (still valid devInfo( isCanceledCache: False, isValidCallback: True, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:43.516) - winInputHook (10632):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (18:10:45.049) - winInputHook (10632):
Input: kb(laptop):shift+f12
IO - tones.beep (18:10:45.053) - MainThread (5080):
Beep at pitch 200, for 100 ms, left volume 50, right volume 50
IO - speech.speak (18:10:45.072) - MainThread (5080):
Speaking ['Visual Studio Code', '8 de 9', CancellableSpeech (still valid devInfo( isCanceledCache: False, isValidCallback: True, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:45.265) - winInputHook (10632):
Input: kb(laptop):shift+f11
IO - speech.speak (18:10:45.286) - MainThread (5080):
Speaking ['Microsoft Teams', '7 de 9', CancellableSpeech (cancelled devInfo( isCanceledCache: True, isValidCallback: False, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:45.461) - winInputHook (10632):
Input: kb(laptop):shift+f11
IO - speech.speak (18:10:45.489) - MainThread (5080):
Speaking ['WinSCP', '6 de 9', CancellableSpeech (cancelled devInfo( isCanceledCache: True, isValidCallback: False, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:45.620) - winInputHook (10632):
Input: kb(laptop):shift+f11
IO - speech.speak (18:10:45.644) - MainThread (5080):
Speaking ['NVDA', '5 de 9', CancellableSpeech (cancelled devInfo( isCanceledCache: True, isValidCallback: False, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:45.768) - winInputHook (10632):
Input: kb(laptop):shift+f11
IO - speech.speak (18:10:45.784) - MainThread (5080):
Speaking ['Microsoft Edge', '4 de 9', CancellableSpeech (cancelled devInfo( isCanceledCache: True, isValidCallback: False, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:45.943) - winInputHook (10632):
Input: kb(laptop):shift+f11
IO - speech.speak (18:10:45.972) - MainThread (5080):
Speaking ['FolderView', 'lista', CancellableSpeech (still valid devInfo( isCanceledCache: False, isValidCallback: True, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:47.065) - winInputHook (10632):
Input: kb(laptop):shift+f12
IO - speech.speak (18:10:47.084) - MainThread (5080):
Speaking ['Microsoft Edge', '4 de 9', CancellableSpeech (cancelled devInfo( isCanceledCache: True, isValidCallback: False, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:47.197) - winInputHook (10632):
Input: kb(laptop):shift+f12
IO - speech.speak (18:10:47.207) - MainThread (5080):
Speaking ['NVDA', '5 de 9', CancellableSpeech (cancelled devInfo( isCanceledCache: True, isValidCallback: False, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:47.377) - winInputHook (10632):
Input: kb(laptop):shift+f12
IO - speech.speak (18:10:47.395) - MainThread (5080):
Speaking ['WinSCP', '6 de 9', CancellableSpeech (cancelled devInfo( isCanceledCache: True, isValidCallback: False, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:47.565) - winInputHook (10632):
Input: kb(laptop):shift+f12
IO - speech.speak (18:10:47.568) - MainThread (5080):
Speaking ['Microsoft Teams', '7 de 9', CancellableSpeech (cancelled devInfo( isCanceledCache: True, isValidCallback: False, isValidCallbackDevInfo:  )]
IO - inputCore.InputManager.executeGesture (18:10:48.742) - winInputHook (10632):
Input: kb(laptop):shift+f12
IO - speech.speak (18:10:48.763) - MainThread (5080):
Speaking ['Visual Studio Code', '8 de 9', CancellableSpeech (still valid devInfo( isCanceledCache: False, isValidCallback: True, isValidCallbackDevInfo:  )]
```

Thanks

**Comment by fisher729** [comment]
Hi.

This is probably happening because you have attempt to cancel speech set to yes in your configuration. The add-on doesn't work well with that setting enabled.

**Comment by brunoprietog** [comment]
Yes, I accidentally noticed that too. It would be nice if they could fix it, considering this setting is enabled by default in NVDA 2021.1.

**Comment by brunoprietog** [comment]
This is already fixed
